### PR TITLE
fix: csv export code refactor for serializer along with filtration and ordering

### PR DESCRIPTION
**Pull request by tehreem-sadat**
**Update Tasks list view:**
- Add default ordering on ID desc order.
  - related issue: https://github.com/nelc/futurex-openedx-extensions/issues/127 
- Add filtration on view name and related_id (It will be course_id for course learners detail view.
  - related issue: https://github.com/nelc/futurex-openedx-extensions/issues/129
- Use SerializerOptionalMethodField for download_url
- Replace tenant_id from integer field to fk relation with TenantConfig

Example requests:
- Following will return all tasks details along with download_url for course_id "course-v1:nelp_org+222+22" in response
  - `/api/fx/export/v1/tasks/?optional_field_tags=__all__&related_id=course-v1%3Anelp_org%2B222%2B22`

### CODE REVIEW

**Review by OmarIthawi — DISMISSED** [review]
Thanks @tehreem-sadat.

`created_at` isn't indexed, so this will be slow. Please either use `-id` or add index to the `created_at` field.

File: futurex_openedx_extensions/helpers/models.py [hunk 1]
```diff
@@ -306,10 +306,11 @@ class DataExportTask(models.Model):
 
     filename = models.CharField(max_length=255)
     view_name = models.CharField(max_length=255)
+    related_id = models.CharField(max_length=255, null=True, blank=True)
```
Comment (OmarIthawi) [inline futurex_openedx_extensions/helpers/models.py:309]: I'm unsure if it's helpful to have both `blank` and `null`, I think one of them is needed.

@shadinaif @tehreem-sadat does indexing aid search in this field?

```suggestion
    related_id = models.CharField(max_length=255, null=True)
```

File: futurex_openedx_extensions/helpers/models.py [hunk 2]
```diff
@@ -306,10 +306,11 @@ class DataExportTask(models.Model):
...
     user = models.ForeignKey(User, on_delete=models.CASCADE)
     status = models.CharField(max_length=16, choices=STATUS_CHOICES, default=STATUS_IN_QUEUE)
     progress = models.FloatField(default=0.0)
-    notes = models.CharField(max_length=255, default='')
+    notes = models.CharField(max_length=255, default='', blank=True)
```
Comment (OmarIthawi) [inline futurex_openedx_extensions/helpers/models.py:313]: 👍🏼

File: futurex_openedx_extensions/helpers/models.py [hunk 3]
```diff
@@ -306,10 +306,11 @@ class DataExportTask(models.Model):
...
     tenant_id = models.IntegerField()
```
Comment (OmarIthawi) [inline futurex_openedx_extensions/helpers/models.py:314]: This should be a foreign key.

File: futurex_openedx_extensions/helpers/models.py [hunk 4]
```diff
@@ -306,10 +306,11 @@ class DataExportTask(models.Model):
 
     filename = models.CharField(max_length=255)
     view_name = models.CharField(max_length=255)
```
Comment (OmarIthawi) [inline futurex_openedx_extensions/helpers/models.py:308]: @shadinaif @tehreem-sadat does indexing aid search here?

File: tests/test_dashboard/test_views.py [hunk 5]
```diff
@@ -765,6 +765,15 @@ def test_permission_classes(self):
...
         self.assertEqual(view_class.permission_classes, [FXHasTenantCourseAccess])
 
+    def test_get_related_id(self):
+        """Verify get_related_id returns course_id"""
+        view_func, _, kwargs = resolve(self.url)
+        view = view_func.view_class()
+        view.kwargs = kwargs
+        expected_related_id = 'course-v1:ORG1+5+5'
+        related_id = view.get_related_id()
+        self.assertEqual(expected_related_id, related_id)
```
Comment (OmarIthawi) [inline tests/test_dashboard/test_views.py:775]: Please prefer plain asserts for better debugging using `pytest` and slightly faster performance than using old methods.

```suggestion
        assert expected_related_id == related_id
```

**Review by OmarIthawi — COMMENTED** [review]
Few comments, no blockers.

@shadinaif please check and approve if all is good.

File: futurex_openedx_extensions/helpers/models.py (hunk 1 above)
Comment (tehreem-sadat) [reply futurex_openedx_extensions/helpers/models.py:309]: blank=True will help to add/update db model entry from django-admin.

File: futurex_openedx_extensions/helpers/models.py (hunk 4 above)
Comment (tehreem-sadat) [reply futurex_openedx_extensions/helpers/models.py:308]: added.

File: futurex_openedx_extensions/helpers/models.py (hunk 3 above)
Comment (tehreem-sadat) [reply futurex_openedx_extensions/helpers/models.py:314]: We started with foreign key, then there were issues in tests so for simplicity we used integer field. @shadinaif please add your comments here also was it a temporary change and should I work on it now ?

**Review by OmarIthawi — COMMENTED** [review]
I think this will reflect the actual usage patterns.

File: futurex_openedx_extensions/helpers/models.py (hunk 3 above)
Comment (shadinaif) [reply futurex_openedx_extensions/helpers/models.py]: fixed by @tehreem-sadat , with many thanks 🌷

File: test_utils/edx_platform_mocks/django_filters/rest_framework.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    def filter_queryset(
+        self, request, queryset, view
+    ):  # pylint: disable=unused-argument, no-self-use
+        """
+        This is the mock implementation of filter_queryset.
+        In a real scenario, it would filter the queryset based on the request parameters.
+
+        In this mock class, it doesn't filter the queryset at all and just returns it.
+        """
+        return queryset  # Simply return the queryset as-is (no filtering).
```
Comment (shadinaif) [inline test_utils/edx_platform_mocks/django_filters/rest_framework.py:17]: @tehreem-sadat , please remove this mock completely. I've added `django-filter` to [tests requirements](https://github.com/nelc/futurex-openedx-extensions/pull/132). Just rebase on `main` and your tests will pass without this mock

**Review by shadinaif — APPROVED** [review]
See my request please, then squash and merge. Thank you @tehreem-sadat

**Pull request merged**